### PR TITLE
Add UpSampling gradient tests

### PR DESCRIPTION
**Pull request by Shashi456**

### CODE REVIEW

File: Tests/TensorFlowTests/LayerTests.swift [hunk 1]
```diff
@@ -492,6 +492,25 @@ final class LayerTests: XCTestCase {
       XCTAssertEqual(output.shape, expected)
     }
 
+    func testUpsampling1DGradient() {
```
Comment (rxwei) [inline Tests/TensorFlowTests/LayerTests.swift]: Upsampling -> UpSampling for all occurrences.

File: Tests/TensorFlowTests/LayerTests.swift [hunk 2]
```diff
@@ -512,6 +549,25 @@ final class LayerTests: XCTestCase {
...
+        // The expected value of the gradient was computed using the following Python code:
+        // ```
+        //   UpSampling3D = tf.keras.layers.UpSampling3D(size = 3)
+        //   with tf.GradientTape() as t:
+        //     t.watch(x)
+        //     y = tf.math.reduce_sum(Upsampling3D(x))
+        //   print(t.gradient(y, x))
+        // ```
+        XCTAssertEqual(computedGradient.0,
+                      [[[[[27.0, 27.0, 27.0, 27.0], [27.0, 27.0, 27.0, 27.0]],
```
Comment (rxwei) [inline Tests/TensorFlowTests/LayerTests.swift]: Fix indentation
```suggestion
                       [[[[[27.0, 27.0, 27.0, 27.0], [27.0, 27.0, 27.0, 27.0]],
```

**Comment by Shashi456** [comment]
This can be merged.
